OpenRA.Mods.Common/Traits/Buildings/BuildingUtils.cs

```diff
@@ -17,15 +17,25 @@ namespace OpenRA.Mods.Common.Traits
 {
 	public static class BuildingUtils
 	{
-		public static bool IsCellBuildable(this World world, CPos cell, BuildingInfo bi, Actor toIgnore = null)
+		public static bool IsCellBuildable(this World world, CPos cell, ActorInfo ai, BuildingInfo bi, Actor toIgnore = null)
 		{
 			if (!world.Map.Contains(cell))
 				return false;
 
+			// cell is occupied by a structure
 			if (world.WorldActor.Trait<BuildingInfluence>().GetBuildingAt(cell) != null)
-				return false;
-
-			if (!bi.AllowInvalidPlacement && world.ActorMap.GetActorsAt(cell).Any(a => a != toIgnore))
+			{
+				// check for 'Replacement'-trait & if it fits
+				if (ai == null)
+					return false;
+
+				var replacementInfo = ai.TraitInfoOrDefault<ReplacementInfo>();
+				if (!(replacementInfo != null &&
```

Comment:
This would be more clear as replacementInfo == null && !IsBuildingAtCellReplaceable(...

Reply:
simple
```
if (replacementInfo == null)
	return false;
if (!IsBuildingAtCellReplaceable(world, cell, replacementInfo))
	return false;
```
or
```
if ((replacementInfo == null) ? true : !IsBuildingAtCellReplaceable(world, cell, replacementInfo))
	return false;
```